src/v/cluster/bootstrap_backend.cc

```diff
@@ -14,16 +14,31 @@
 #include "cluster/cluster_uuid.h"
 #include "cluster/commands.h"
 #include "cluster/logger.h"
+#include "cluster/members_manager.h"
 #include "cluster/types.h"
 #include "security/credential_store.h"
 
 namespace cluster {
 
+// node_ids_by_uuid initialization
+// ===============================
+// If the node UUID to ID map in `members_manager` were not initialized at
+// cluster bootstrap, that would have allowed slow seed nodes (ones that miss
+// their oportunity to be founding nodes because of being slow) to also possibly
+// miss the assignment of `node_id` based on `seed_servers` index (as expected
+// for seed nodes). That could happen if another node joins the cluster before a
+// slow seed node and grabs the `node_id` that should have belonged to it.
+//
+// `bootstrap_backend` references `members_manager` to be able to initialize the
+// node UUID to ID map in it.
```

Comment:
nit: IMO kinda odd placement; not sure folks would know to look for this context here. Maybe move this down by L146? or somewhere in the header?